src/test/java/org/cryptomator/common/PropertiesPreprocessorTest.java

```diff
@@ -0,0 +1,24 @@
+package org.cryptomator.common;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.Mockito;
+
+public class PropertiesPreprocessorTest {
+
+	@ParameterizedTest
+	@CsvSource(value = """
+			org.example.@{mytest1}.test, org.example.@{mytest1}.test
+			@{only*word*digits*under_score\\},@{only*words*digits*under_score\\}
+			C:\\Users\\@{appdir}\\dir, C:\\Users\\\\dir
+			@{@{appdir}},@{}
+			Longer @{appdir} text with @{appdir}., Longer  text with .
+			""")
+	public void test(String propertyValue, String expected) {
+		var result = PropertiesPreprocessor.process(propertyValue);
+		Assertions.assertEquals(result, expected);
+	}
```

Comment:
This test has no control over the value of `appdir`, making its result system-dependent.

Can you instead add an overloaded function for `process` which accepts a second argument (the environment), which can get mocked?

Reply:
Fixed in https://github.com/cryptomator/cryptomator/pull/2957/commits/ebc60c4fb37421442e6b23fa3f3f38c8204b26a3